### ISSUE TITLE
Use TestPurpose in LuckyPeople

### ISSUE THREAD

**Issue by keremispirli** [TorXakis/TorXakis#173]
- [x] Replace old style purpose models with proper Test Purposes
- [x] Add a test using IncrementRows Purpose to test for luck based on previous people's genders (MMMMMF or FFFFFM) (#157)
- [x] Update documentation - [Modelling Example Lucky People (Solution)](https://github.com/TorXakis/TorXakis/wiki/Modelling-Example-Lucky-People-(Solution))

**Comment by pjljvandelaar** [comment]
Since Jan recently has develop Test Purpose, I want to test some Test Purposes as soon as possible.
The label low-priority is thus inappropriate according to me....

**Comment by keremispirli** [comment]
Since #44 is fixed, I tried this again and looks like using Test Purpose for examples work:

```
TXS >>  LuckyPeople.txs
TXS <<  tester Model PurposeExamples Sut
TXS >>  Tester started
TXS <<  test 50
TXS >>  .....1: IN: Act { { ( In, [ Person(Male,"Mickey","Mouse",13,1) ] ) } }
TXS >>  .....2: OUT: Act { { ( Out, [ True ] ) } }
TXS >>  .....3: OUT: No Output (Quiescence)
TXS >>  .....4: IN: Act { { ( In, [ Person(Male,"Donald","Duck",13,3) ] ) } }
TXS >>  .....5: OUT: Act { { ( Out, [ True ] ) } }
TXS >>  .....6: OUT: No Output (Quiescence)
TXS >>  .....7: IN: Act { { ( In, [ Person(Male,"Luuk","Laar",24,12) ] ) } }
TXS >>  .....8: OUT: Act { { ( Out, [ True ] ) } }
TXS >>  .....9: OUT: No Output (Quiescence)
TXS >>  ....10: IN: Act { { ( In, [ Person(Female,"Shakira","Ripoll",2,2) ] ) } }
TXS >>  ....11: OUT: Act { { ( Out, [ True ] ) } }
TXS >>  ....12: IN: Act { { ( In, [ Person(Male,"Michael","Buble",9,9) ] ) } }
TXS >>  ....13: OUT: Act { { ( Out, [ True ] ) } }
TXS >>  ....14: OUT: No Output (Quiescence)
TXS >>  ....15: IN: Act { { ( In, [ Person(Female,"Imke","Laar",7,7) ] ) } }
TXS >>  ....16: OUT: Act { { ( Out, [ True ] ) } }
TXS >>  ....17: OUT: No Output (Quiescence)
TXS >>  ....18: IN: Act { { ( In, [ Person(Male,"Huey","Duck",17,10) ] ) } }
TXS >>  ....19: OUT: Act { { ( Out, [ False ] ) } }
TXS >>  ....20: OUT: No Output (Quiescence)
TXS >>  ....21: IN: Act { { ( In, [ Person(Male,"Dewey","Duck",17,10) ] ) } }
TXS >>  ....22: OUT: Act { { ( Out, [ True ] ) } }
TXS >>  ....23: IN: Act { { ( In, [ Person(Male,"Louie","Duck",17,10) ] ) } }
TXS >>  ....24: OUT: Act { { ( Out, [ False ] ) } }
TXS >>  ....25: IN: Act { { ( In, [ Person(Male,"Mickey","Mouse",13,1) ] ) } }
TXS >>  ....26: OUT: Act { { ( Out, [ True ] ) } }
TXS >>  ....27: OUT: No Output (Quiescence)
TXS >>  ....28: IN: Act { { ( In, [ Person(Male,"Donald","Duck",13,3) ] ) } }
TXS >>  ....29: OUT: Act { { ( Out, [ True ] ) } }
TXS >>  ....30: IN: Act { { ( In, [ Person(Female,"April","Duck",15,5) ] ) } }
TXS >>  ....31: OUT: Act { { ( Out, [ True ] ) } }
TXS >>  ....32: OUT: No Output (Quiescence)
TXS >>  ....33: IN: Act { { ( In, [ Person(Female,"Beatrix","Oranje",31,1) ] ) } }
TXS >>  ....34: OUT: Act { { ( Out, [ False ] ) } }
TXS >>  ....35: IN: Act { { ( In, [ Person(Female,"Maxima","Zorreguieta",17,5) ] ) } }
TXS >>  ....36: OUT: Act { { ( Out, [ False ] ) } }
TXS >>  ....37: IN: Act { { ( In, [ Person(Female,"Amalia","Oranje",7,12) ] ) } }
TXS >>  ....38: OUT: Act { { ( Out, [ False ] ) } }
TXS >>  ....39: OUT: No Output (Quiescence)
TXS >>  ....40: IN: Act { { ( In, [ Person(Female,"Alexia","Oranje",26,6) ] ) } }
TXS >>  ....41: OUT: Act { { ( Out, [ False ] ) } }
TXS >>  ....42: OUT: No Output (Quiescence)
TXS >>  ....43: IN: Act { { ( In, [ Person(Female,"Ariane","Oranje",10,4) ] ) } }
TXS >>  ....44: OUT: Act { { ( Out, [ False ] ) } }
TXS >>  ....45: IN: Act { { ( In, [ Person(Male,"Willem","Oranje",27,4) ] ) } }
TXS >>  ....46: OUT: Act { { ( Out, [ True ] ) } }
TXS >>  ....47: OUT: No Output (Quiescence)
TXS >>  Goal examples: Hit
TXS >>  PASS
```

But despite having exactly same process with Overflow model, PurposeOverflow halts after a single test while the model continues testing as long as you want:
```
TXS <<  tester Model PurposeOverflow Sut
TXS >>  Tester started
TXS <<  test 100
TXS >>  .....1: IN: Act { { ( In, [ Person(Male,"Lp","C",26,11) ] ) } }
TXS >>  .....2: OUT: Act { { ( Out, [ False ] ) } }
TXS >>  .....3: OUT: No Output (Quiescence)
TXS >>  Goal overflow: halted
TXS >>  PASS
TXS <<  test 100
TXS >>  .....1: OUT: No Output (Quiescence)
TXS >>  Goal overflow: halted
TXS >>  PASS
```

@pjljvandelaar Any ideas?

**Comment by keremispirli** [comment]
Blocked by #201

**Comment by keremispirli** [comment]
@pjljvandelaar @capitanbatata
Could you comment on the changes in documentation?
[Modelling Example Lucky People (Solution)](https://github.com/TorXakis/TorXakis/wiki/Modelling-Example-Lucky-People-(Solution))
[Changes (1f16eb1)](https://github.com/TorXakis/TorXakis/wiki/Modelling-Example-Lucky-People-(Solution)/_compare/cb0daa069d7a2054af84ca3ab5741194c36dcf06...1f16eb1074875a18f29c4e2d3e3818e1ced2c435)

**Comment by keremispirli** [comment]
I updated the documentation based on latest Test Purposes and split it into pages.
[Modelling Example Lucky People (Solution)](https://github.com/TorXakis/TorXakis/wiki/Modelling-Example-Lucky-People-(Solution))
